src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.service.impl;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.Order;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.OrderService;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        if (shoppingCart == null) {
+            throw new IllegalArgumentException("shoppingCart is null");
+        }
+        if (shoppingCart.getTickets() == null || shoppingCart.getTickets().isEmpty()) {
+            throw new IllegalArgumentException("tickets is null or empty");
```

Comment:
The check for null or empty tickets could be improved by using `Objects.requireNonNull(shoppingCart.getTickets(), "tickets is null")` and then checking if the list is empty separately.